pkg/manager/impl/execution_manager.go

```diff
@@ -465,6 +465,30 @@ func (m *ExecutionManager) getExecutionConfig(ctx context.Context, request *admi
 	}, nil
 }
 
+func (m *ExecutionManager) getClusterAssignment(ctx context.Context, request *admin.ExecutionCreateRequest) (
+	*admin.ClusterAssignment, error) {
+	if request.Spec.ClusterAssignment != nil {
```

Comment:
Do we need a nil check on spec too

Reply:
nope! it's already been validated at this point: https://github.com/flyteorg/flyteadmin/blob/master/pkg/manager/impl/validation/execution_validator.go#L49